feature-settings-impl/src/main/java/io/novafoundation/nova/feature_settings_impl/presentation/cloudBackup/CloudBackupStateMapper.kt

```diff
@@ -0,0 +1,107 @@
+package io.novafoundation.nova.feature_settings_impl.presentation.cloudBackup
+
+import io.novafoundation.nova.common.resources.ResourceManager
+import io.novafoundation.nova.common.utils.formatting.formatDateSinceEpoch
+import io.novafoundation.nova.common.utils.formatting.formatTime
+import io.novafoundation.nova.feature_cloud_backup_api.domain.model.errors.CannotApplyNonDestructiveDiff
+import io.novafoundation.nova.feature_cloud_backup_api.domain.model.errors.CloudBackupAuthFailed
+import io.novafoundation.nova.feature_cloud_backup_api.domain.model.errors.CloudBackupNotEnoughSpace
+import io.novafoundation.nova.feature_cloud_backup_api.domain.model.errors.CloudBackupUnknownError
+import io.novafoundation.nova.feature_cloud_backup_api.domain.model.errors.CloudBackupWrongPassword
+import io.novafoundation.nova.feature_cloud_backup_api.domain.model.errors.CorruptedBackupError
+import io.novafoundation.nova.feature_settings_impl.R
+import java.util.Date
+
+fun mapCloudBackupStateModel(
+    resourceManager: ResourceManager,
+    backupEnabled: Boolean,
+    syncingInProgress: Boolean,
+    errorState: Throwable?,
+    lastSync: Date?
+): CloudBackupStateModel {
+    val (stateImage, stateImageTint, stateBackgroundColor) = mapStateImage(backupEnabled, errorState)
+
+    return CloudBackupStateModel(
+        stateImg = stateImage,
+        stateImageTint = stateImageTint,
+        stateColorBackgroundRes = stateBackgroundColor,
+        showProgress = syncingInProgress,
+        title = mapCloudBackupStateTitle(resourceManager, backupEnabled, syncingInProgress, errorState),
+        subtitle = mapCloudBackupStateSubtitle(resourceManager, backupEnabled, lastSync),
+        isClickable = mapCloudBackupClickability(backupEnabled, syncingInProgress, errorState),
+        problemButtonText = mapCloudBackupProblemButton(resourceManager, backupEnabled, syncingInProgress, errorState)
+    )
```

Comment:
Imho it is better to have a mapper that creates the whole state, not multiple separate partical ones - that makes cases harder to understand since they are splitted